opentelemetry_collector/receiver/dockerstats/scraper.go

```diff
@@ -0,0 +1,150 @@
+package dockerstats
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/golang/glog"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/consumer/pdatautil"
+
+	mpb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+
+	"github.com/googlecloudplatform/appengine-sidecars-docker/opentelemetry_collector/receiver/metricgenerator"
+)
+
+var (
+	containerNameLabel = &mpb.LabelKey{
+		Key:         "container_name",
+		Description: "Name of the container (or ID if name is not available)",
+	}
+
+	memUsageDesc = &mpb.MetricDescriptor{
+		Name:        "container/memory/usage",
+		Description: "Total memory the container is using",
+		Unit:        "By",
+		Type:        mpb.MetricDescriptor_GAUGE_INT64,
+		LabelKeys:   []*mpb.LabelKey{containerNameLabel},
+	}
+	memLimitDesc = &mpb.MetricDescriptor{
+		Name:        "container/memory/limit",
+		Description: "Total memory the container is allowed to use",
+		Unit:        "By",
+		Type:        mpb.MetricDescriptor_GAUGE_INT64,
+		LabelKeys:   []*mpb.LabelKey{containerNameLabel},
+	}
+)
+
+type scraper struct {
+	startTime      time.Time
+	scrapeInterval time.Duration
+	done           chan bool
+
+	metricConsumer consumer.MetricsConsumer
+	docker         client.ContainerAPIClient
+}
+
+func newScraper(scrapeInterval time.Duration, metricConsumer consumer.MetricsConsumer) (*scraper, error) {
+	docker, err := client.NewEnvClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize docker client: %v", err)
+	}
+
+	return &scraper{
+		scrapeInterval: scrapeInterval,
+		done:           make(chan bool),
+		metricConsumer: metricConsumer,
+		docker:         docker,
+	}, nil
+}
+
+func (s *scraper) start() {
+	s.startTime = time.Now()
+	go func() {
+		ticker := time.NewTicker(s.scrapeInterval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				s.export()
+			case <-s.done:
+				return
+			}
+		}
+	}()
+}
+
+func (s *scraper) stop() {
+	s.done <- true
+}
+
+func (s *scraper) export() {
```

Comment:
Should we include a log line saying we're starting an export here? I'm thinking about debugging the case where none or not all of the containers are showing up.

Reply:
I've a minor concern with log spam -- any log line here would be printed every 30s. Can you explain more what particular scenario would you want to be able to debug? Maybe there's a better place for such a line.

Reply:
If we don't see any metrics showing up, then something with docker or OpenTelemetry is up. How can we tell if OpenTelemetry is the problem?

Reply:
I see what you're saying. Added an info log line.